sys-boot/udk/udk-2018-r1.ebuild

```diff
@@ -0,0 +1,204 @@
+	sed -e "s:^\(BUILD_CFLAGS\s*=\).*$:\1 ${CFLAGS} -MD -fshort-wchar -fno-strict-aliasing -nostdlib -c -fPIC:" \
+		-i "BaseTools/Source/C/Makefiles/header.makefile" \
+		|| die "Failed to update makefile header"
+}
+
+src_compile() {
+	local make_flags=(
+		BUILD_CC="$(tc-getBUILD_CC)"
+		BUILD_CXX="$(tc-getBUILD_CXX)"
+		BUILD_AS="$(tc-getBUILD_AS)"
+		BUILD_AR="$(tc-getBUILD_AR)"
+		BUILD_LD="$(tc-getBUILD_LD)"
+	)
+	# Base tools does not like parallel make
+	emake "${make_flags[@]}" -j1 -C BaseTools
+
+	# Update template parameter files
+	sed -e "s:^\(ACTIVE_PLATFORM\s*=\).*$:\1 MdeModulePkg/MdeModulePkg.dsc:" \
+		-e "s:^\(TARGET\s*=\).*$:\1 RELEASE:" \
+		-e "s:^\(TARGET_ARCH\s*=\).*$:\1 ${EFIARCH}:" \
+		-e "s:^\(TOOL_CHAIN_TAG\s*=\).*$:\1 ${TOOLCHAIN_TAG}:" \
+		-e "s:^\(MAX_CONCURRENT_THREAD_NUMBER\s*=\).*$:\1 $(makeopts_jobs):" \
+		-i "BaseTools/Conf/target.template" || die "Failed to configure target file"
+
+	# Clean unneeded files
+	find . -name '*.bat' -exec rm -f {} \;
+	find ./BaseTools/Source/C -mindepth 1 -maxdepth 1 \! -name 'bin' -exec rm -rf {} \;
+
+	# Create workspace script file
+	cat >>"${S}/udk-workspace" <<-EOF \
+		|| die "Failed to create workspace script"
+		#!/bin/bash
+
+		show_usage() {
+		  if [ ! -z "\${1}" ]; then
+		    echo "\${1}" >&2
+		    echo >&2
+		  fi
+		  echo "Usage:"
+		  echo ". \${BASH_SOURCE[0]} [folder]"
+		  echo "where folder is the workspace folder - current directory is used if unspecified"
+		}
+
+		create_workspace() {
+		  local workspace_dir="\${PWD}"
+		  if [ ! -z "\${1}" ]; then
+		    workspace_dir="\${1}"
+		    if [ ! -d "\${1}" ]; then
+		    if ! mkdir --parent "\${1}"; then
+		      show_usage "Cannot create directory \${1}"
+		      return 1
+		    fi
+		    fi
+		  fi
+
+		  pushd "\${workspace_dir}" >>/dev/null
+		  if [ ! -d Conf ] && ! mkdir Conf; then
+		    show_usage "Cannot create configuration directory"
+		    return 1
+		  fi
+		  export WORKSPACE="\${PWD}"
+		  popd >>/dev/null
+		}
+
+		if [ "\${1}" = "-h" ] || [ "\${1}" = "--help" ]; then
+		  show_usage
+		  return 2>>/dev/null
+		  exit 0
+		elif [ "\${BASH_SOURCE[0]}" = "\${0}" ]; then
+		  show_usage "Script must be sourced"
+		  exit 1
+		elif [ \$# -gt 1 ]; then
+		  show_usage "Too many arguments"
+		  return 1
+		fi
+
+		if ! create_workspace "\${1}"; then
+		  return 1
+		fi
+		shift
+		unset -f show_usage
+		unset -f create_workspace
+
+		export EDK_TOOLS_PATH="${EPREFIX}/usr/share/${P}/BaseTools"
+		. "${EPREFIX}/usr/share/${P}/edksetup.sh"
+
+		echo "Do not forget to call this script each time you open a new terminal or work on"
+		echo "another workspace."
+		echo "For more information, type:"
+		echo "\${BASH_SOURCE[0]} --help"
+	EOF
+}
+
+src_install() {
+	exeinto "/usr/bin"
+	doexe udk-workspace
+	rm -f udk-workspace
+
+	while read -d '' -r filename; do
+		exeinto "/usr/share/${P}/${filename%/*}"
+		doexe "${filename}"
+		rm -f "${filename}"
+	done < <(find . -executable -type f -print0)
```

Comment:
`doins -r` should preserve executable bits, you don't need to hack it around.

Reply:
Actually, I did the hack because ’doins -r’ did not preserve executable bits, when I did my tests. Is there a flag or something I should give?

Reply:
You are correct, it doesn't. Which is kinda silly.

I think the simplest solution would be to replace this whole block with plain `cp -R -p`.